teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -144,6 +144,8 @@ interface BuildLocator {
      */
     fun withAllBranches(): BuildLocator
 
+    fun withDefaultFilter(enabled: Boolean): BuildLocator
```

Comment:
Meaning of this function is unclear, I needed to look into TeamCity REST API doc to understand what it is about. One of the goals of this library is to provided clear API which can be used without having to look into REST API doc (see [CONTRIBUTING.md](https://github.com/JetBrains/teamcity-rest-client/blob/master/CONTRIBUTING.md)).

Well, we can put text from the documentation to KDoc here, but it doesn't help much. When users need to invoke this function? You wrote "The defaultFilter hides some builds and there was no way to find them previously". Could you please provide more details which builds weren't included and which options in BuildLocator did you use? E.g. if in order to show cancelled or personal builds we need to set `defaultFilter` to `false`, we may do this automatically in `BuildLocatorImpl::all` without adding this function to API.

Reply:
Well I don't really know how to describe what default filter does as the logic behind it can change, but the point is that it hides the builds from user even you made all possible calls in the builder, so the option is very much needed. 

Regarding adding it by default - I'm all for that, but that changes the previous behavior and after the update, users may be surprised as to why do they get more builds.

To reiterate: I'll gladly make it default. Do you think that we need a method like `withDefaultFilter()` which will just turn it on?

Reply:
```
val dependencies = teamCityInstance.builds()
                .includeFailed()
                .includeCanceled()
                .withAllBranches()
                .snapshotDependencyTo(build.id)
                .all()
                .toList()
```
this is what I used, and IIRC the builds that failed to start were not included